modules/transport/Transport.js

```diff
@@ -0,0 +1,191 @@
+
+            if (listener) {
+                if (this._transport) {
+                    this._transport.removeListener(eventName, listener);
+                }
+                this._transportListeners.delete(eventName);
+            }
+        });
+    }
+
+    /**
+     * Disposes the current transport.
+     *
+     * @returns {void}
+     */
+    _disposeTransport() {
+        if (this._transport) {
+            this._transport.dispose();
+            this._transport = null;
+        }
+    }
+
+    /**
+     * Handles incomming data from the transport.
+     *
+     * @param {string} name - The name of the event, request or response
+     * related to the data received.
+     * @param {Object} data - The data.
+     * @returns {void}
+     */
+    _onDataReceived(name, data) {
+        if (data.type === MESSAGE_TYPE_RESPONSE) {
+            // will be handled in sendRequest method.
+            return;
+        }
+        let callback;
+
+        if (data.type === MESSAGE_TYPE_REQUEST) {
+            callback = (result, error) => {
+                this._transport.send(name, {
+                    type: MESSAGE_TYPE_RESPONSE,
+                    result,
+                    error,
+                    id: data.id
+                });
+            };
+        } else {
+            callback = () => {
+                // no need to send response.
+            };
+        }
+        this.emit(name, data.data, callback);
+    }
+
+    /**
+     * Overrides {@link EventEmitter#addListener()} to also add listener to
+     * the low level transport class.
+     *
+     * @inheritdoc
+     * @override
+     */
+    addListener(eventName, listener) {
+        super.addListener(eventName, listener);
+        const transportListener = this._onDataReceived.bind(this, eventName);
+
+        this._transportListeners.set(eventName, transportListener);
+        if (this._transport) {
+            this._transport.on(eventName, transportListener);
+        }
+    }
+
+    /**
+     * Disposes the current transport and removes all listeners.
+     *
+     * @returns {void}
+     */
+    dispose() {
+        this.removeAllListeners();
+        this._disposeTransport();
+    }
+
+    /**
+     * Sends the passed data.
+     *
+     * @param {string} name - The name of the event.
+     * @param {Object} data - The data to be sent.
+     * @returns {void}
+     */
+    sendEvent(name, data = {}) {
+        if (this._transport) {
+            this._transport.send(name, {
+                type: MESSAGE_TYPE_EVENT,
+                data
+            });
+        }
+    }
+
+    /**
+     * Sending request.
+     *
+     * @param {string} name - The name of the request.
+     * @param {Object} data - The data for the request.
+     * @returns {Promise}
+     */
+    sendRequest(name, data) {
+        if (!this._transport) {
+            return Promise.reject(new Error('No transport defined!'));
+        }
+        const id = this._requestID;
```

Comment:
Hmm. My idea was to increase this on every request. 

I have to check your proposal. But isn't the increasing by 1 good enough for the initial implementation?